pkg/storage/tsdb/config.go

```diff
@@ -355,8 +355,16 @@ type BucketStoreConfig struct {
 	// Controls experimental options for index-header file reading.
 	IndexHeader indexheader.Config `yaml:"index_header" category:"experimental"`
 
-	StreamingBatchSize   int `yaml:"streaming_series_batch_size" category:"advanced"`
-	ChunkRangesPerSeries int `yaml:"fine_grained_chunks_caching_ranges_per_series" category:"experimental"`
+	StreamingBatchSize          int    `yaml:"streaming_series_batch_size" category:"advanced"`
+	ChunkRangesPerSeries        int    `yaml:"fine_grained_chunks_caching_ranges_per_series" category:"experimental"`
+	SeriesSelectionStrategyName string `yaml:"series_selection_strategy" category:"experimental"`
+}
+
+var validSeriesSelectionStrategies = []string{
+	"speculative",
+	"worstCase",
+	"worstCaseSmallPostingLists",
+	"all",
```

Comment:
These should be consts.

Reply:
Also, as we spoke in private (writing the comment in public in case someone disagrees), I think that we should make these dash-separated rather than camelCase (this kind of triggered my OCD).